File: codingjs/mixString.js
function mixString(a: string, b: string): string {
  return mixStringHelper(a, b, "")
}
function mixStringHelper(a: string, b: string, result: string): string {
  if (a === "" || b === "") return result + a + b;
  result += (a.substr(0, 1) + b.substr(0, 1))
  return mixStringHelper(a.substring(1), b.substring(1), result)
}

File: codingbat/java/mixString.java
public String mixString(String a, String b) {
  int shortLen = Math.min(a.length(), b.length());
  String result = "";
  for (int i = 0; i < shortLen; i++) {
    result += a.charAt(i) + "" + b.charAt(i);
  }
  return result + a.substring(shortLen) + b.substring(shortLen);
}

